buildSrc/src/main/java/org/elasticsearch/gradle/ElasticsearchJavaPlugin.java

```diff
@@ -464,31 +461,4 @@ private static void configureJavadoc(Project project) {
         // ensure javadoc task is run with 'check'
         project.getTasks().named(LifecycleBasePlugin.CHECK_TASK_NAME).configure(t -> t.dependsOn(javadoc));
     }
-
-    static class TestFailureReportingPlugin implements Plugin<Project> {
-        @Override
-        public void apply(Project project) {
-            if (project != project.getRootProject()) {
-                throw new IllegalStateException(this.getClass().getName() + " can only be applied to the root project.");
-            }
-
-            project.getGradle().addListener(new TaskActionListener() {
```

Comment:
removing the task listener also fixes an incompatibility with upcoming gradle configuration cache. see https://docs.gradle.org/6.6-rc-1/userguide/configuration_cache.html#config_cache:requirements:build_listeners for details